contracts/erc1155/onchain/OnchainFileStorage.sol

```diff
@@ -0,0 +1,153 @@
+     */
+    bytes[] private _files;
+
+    /**
+     * @notice Add a file via its name and associated storage bytecode addresses
+     */
+    function addFile(string calldata fileName, address[] calldata fileStorageAddresses) external onlyOwner {
+        bytes memory _fileName = bytes(fileName);
+        if (_fileStorage[_fileName].length != 0) {
+            _revert(FileAlreadyRegistered.selector);
+        }
+
+        _files.push(_fileName);
+        _fileStorage[_fileName] = fileStorageAddresses;
+    }
+
+    /**
+     * @notice Remove a file from registered list of file names, and its associated storage bytecode addresses
+     */
+    function removeFile(string calldata fileName) external onlyOwner {
+        bytes memory _fileName = bytes(fileName);
+        if (_fileStorage[_fileName].length == 0) {
+            _revert(FileNotRegistered.selector);
+        }
+
+        bytes[] memory oldFiles = _files;
+        bytes[] memory newFiles = new bytes[](oldFiles.length - 1);
+        uint256 fileIndexOffset = 0;
+        uint256 oldFilesLength = oldFiles.length;
+
+        for (uint256 i = 0; i < oldFilesLength; i++) {
+            if (keccak256(oldFiles[i]) == keccak256(_fileName)) {
+                fileIndexOffset = 1;
+            } else {
+                newFiles[i - fileIndexOffset] = oldFiles[i];
+            }
+        }
+
+        _files = newFiles;
+        delete _fileStorage[_fileName];
+    }
+
+    /**
+     * @notice Return registered file names
+     */
+    function files() external view returns (string[] memory) {
+        bytes[] memory fileNames = _files;
+        string[] memory fileNamesHumanReadable = new string[](fileNames.length);
+
+        for (uint256 i = 0; i < fileNames.length; i++) {
+            fileNamesHumanReadable[i] = string(fileNames[i]);
+        }
+
+        return fileNamesHumanReadable;
+    }
+
+    /**
+     * @notice Return storage bytecode addresses for a file
+     */
+    function fileStorage(string calldata fileName) external view returns (address[] memory) {
+        bytes memory _fileName = bytes(fileName);
+        if (_fileStorage[_fileName].length == 0) {
+            _revert(FileNotRegistered.selector);
+        }
+
+        return _fileStorage[bytes(fileName)];
+    }
+
+    /**
+     * @notice Return file contents
+     */
+    function fileContents(string calldata fileName) external view returns (string memory) {
+        bytes memory _fileName = bytes(fileName);
+        if (_fileStorage[_fileName].length == 0) {
+            _revert(FileNotRegistered.selector);
+        }
+
+        address[] memory fileStorageAddresses = _fileStorage[bytes(fileName)];
+        uint256 fileStorageAddressesLength = fileStorageAddresses.length;
+        string memory contents = "";
+
+        for (uint256 i = 0; i < fileStorageAddressesLength; i++) {
+            contents = string(
+                abi.encodePacked(
```

Comment:
This contract has actually been updated to be more efficient. Overall, this repository is lagging a little behind but this should be the only part that affects your branch. More to come on this later, but expect `main` to be updated with a newer `OnchainFileStorage` contract next week